app/models/manageiq/providers/ansible_tower/configuration_manager/refresh_worker/parser.rb

```diff
@@ -0,0 +1,77 @@
+class ManageIQ::Providers::AnsibleTower::ConfigurationManager::RefreshWorker::Parser
+  attr_accessor :inventory, :ems
+
+  def initialize(ems, target, inventory)
+    @ems    = ems
+    @target = target
+    @inventory = inventory
+  end
+
+  def parse
+    [
+      inventory_groups,
+      configured_systems,
+      configuration_scripts,
+    ]
+  end
+
+  def inventory_groups
+    # FIXME: memoization  should not be needed.
+    @inventory_groups ||= ManagerRefresh::InventoryCollection.new(
+      ManageIQ::Providers::ConfigurationManager::InventoryRootGroup,
+      :association => :inventory_root_groups,
+      :parent      => ems,
```

Comment:
@Ladas see above, if this collection would be s child of an InventoryObject this parent could be infered from that